solutions/src/main/java/ru/sberbank/school/task02/Calculator.java

```diff
@@ -0,0 +1,42 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+public class Calculator implements FxConversionService {
+    private ExternalQuotesService provider;
+
+    Calculator(ExternalQuotesService externalQuotesService) {
+        this.provider = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+        if (amount == null || amount.compareTo(BigDecimal.ZERO) <= 0) {
+            return null;
+        }
+        List<Quote> quotes = provider.getQuotes(symbol);
+        if (quotes.isEmpty()) {
+            return null;
+        }
+
+        Quote current = null;
+        for (Quote quote: quotes) {
+            if (quote.isInfinity() && (current == null)) {
+                current = quote;
+            } else if (amount.compareTo(quote.getVolumeSize()) < 0) {
+                if ((current == null) || current.isInfinity() || (current.getVolumeSize().compareTo(quote.getVolumeSize()) > 0)) {
```

Comment:
Line is longer than 120 characters (found 130).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a617055083>*